### ISSUE TITLE
Nested LTR/RTL-isolates give wrong levels

### ISSUE THREAD

**Issue by poire-z** [fribidi/fribidi#130]
(Not super certain it's a fribidi bug, or me using it wrong - and as I don't read arabic, I'm just looking at the numbers/levels.)

When using nested LRI / RLI, fribidi gives different results than the [Unicode online reference tool](https://unicode.org/cldr/utility/bidic.jsp).

Sample input string in this file: [bidi_nested_isolates.txt](https://github.com/fribidi/fribidi/files/3956199/bidi_nested_isolates.txt)
Which Notepad.exe would show as this (which looks wrong, but the bidi isolate control chars are at the positions I put them):

<kbd>![image](https://user-images.githubusercontent.com/24273478/70720639-75130480-1cf4-11ea-9d8e-2622343f7a7e.png)</kbd>

Fribidi level results (from my code) on the left in black, compared to the output from [https://unicode.org/cldr/utility/bidic.jsp](https://unicode.org/cldr/utility/bidic.jsp?s=%E2%81%A6%E2%81%A7%D8%A7%D9%84%D8%A8%D8%AA%D8%B1%D8%A7%D8%A1+%5BPetra%5D+%28%D8%A7%D9%84%D9%85%D9%85%D9%84%D9%83%D8%A9+%D8%A7%D9%84%D8%A3%D8%B1%D8%AF%D9%86%D9%8A%D8%A9+%D8%A7%D9%84%D9%87%D8%A7%D8%B4%D9%85%D9%8A%D8%A9%29+2017.AR%E2%81%A9%E2%81%A6.epub%E2%81%A9%E2%81%A9&b=1&u=110&d=2) (on the right) - we get different levels:
<kbd>![image](https://user-images.githubusercontent.com/24273478/70714027-928da180-1ce7-11ea-95af-5dbca4a88d63.png)</kbd>

I get the same "different/wrong" levels with fribidi CLI with that input string

```
./fribidi --novisual --nopad --showinput --rtl --levels ~/bidi_nested_isolates.txt
=> 1 2 3 3 3 3 3 3 3 3 4 4 4 4 4 4 4 4 4 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 3 4 3 4 4 4 4 3 4 4 2 2 4 4 4 4 4 1 1
```
(nine "4" at positions 11, while we should only have 5 of them.

My code would then give the following final output:
<kbd>![image](https://user-images.githubusercontent.com/24273478/70714631-d208bd80-1ce8-11ea-8ca2-617ccad1d361.png)</kbd>

While Firefox would show it as expected with `<span dir=>` or bidi control chars:
<kbd>![image](https://user-images.githubusercontent.com/24273478/70714325-370fe380-1ce8-11ea-9f8e-03b3a9af0701.png)</kbd>

(More context about my aim in https://github.com/koreader/koreader/issues/5359#issuecomment-564767926.)

**Comment by dov** [comment]
Thanks for the bug report. I can confirm that this is indeed bug in the fribidi algo. Thanks for doing the initial comparison with the UBA reference.

**Comment by dov** [comment]
Fixed in c7d6e16 . Fribidi now outputs the same result as the UBA reference for the file bidi_nested_isolates.txt .
